python-frontend/src/test/java/org/sonar/python/types/v2/TypesTestUtils.java

```diff
@@ -54,6 +54,13 @@ public static FileInput parseAndInferTypes(PythonFile pythonFile, String... code
     var symbolTable = new SymbolTableBuilderV2(fileInput)
       .build();
     new TypeInferenceV2(PROJECT_LEVEL_TYPE_TABLE, pythonFile, symbolTable).inferTypes(fileInput);
+    return parseAndInferTypes(PROJECT_LEVEL_TYPE_TABLE, pythonFile, code);
```

Comment:
I don't quite understand why the code needs to be parsed and run through type-inference twice

Reply:
That's a very good point, thanks! It's actually a refactoring mistake on my end. I fixed it (and squashed it in the 1st commit).